### ISSUE TITLE
[ENHANCEMENT] Search for specific tag name

### ISSUE THREAD

**Issue by Jos512** [Zettlr/Zettlr#845]
## Description
I have several tags that have similar names, for instance: `#trading` (the activity) and `#tradingview` (a platform for trading).

But if I search for `#trading`, I also get the `#tradingview` results. I don't want that. Else I would have typed `#tradingview` or, if I wanted both, `#trading #tradingview`.

Even if I wrap the tag in double quotes, so `"#trading"`, Zettlr stills shows me results from `#tradingview`.

## Proposed Changes
Have Zettlr perform a literal tag search when the user types the tag name in double quotes, much like how Google searching works.

## Caveats
Not that I can think of; the old search behaviour can still remain when people don't use double quotes.

## Additional Information
- I use version 1.6.0.
- I searched the issues for similar ones but didn't find any. Apologies if I repeat things already requested.
- I searched the changelog for 1.7.0-beta but didn't find a change to how tag searching works, hence me making the issue here. :slightly_smiling_face:

**Comment by nathanlesage** [comment]
Thanks for opening up that issue! In fact, Zettlr will specifically try to match the tags, but it can very well be that it'll just try to match more of the files if no tags match, so this sounds pretty reasonable!

**Comment by Johennes** [comment]
@Jos512 I ran into this, too, and I found out that when you use keywords in the YAML frontmatter, the tag search works as you'd expect it to.

E.g. let's say you have

```
---
keywords:
  - tradingview
---

...
```

then searching for `#tradingview` will return the file in the results but searching for `#trading` will not. You don't even need to wrap it in quotes.

It would be great if it worked the same for inline tags defined with `#tagname`.

**Comment by nathanlesage** [comment]
The reason why in the latter case `#trading` will also be found is that, after the tags themselves have been exhausted Zettlr will start to search the full file contents and treat the tag as a simple string.

I might be able to restrict `#[a-z0-9]` to tags-only, but then I'm afraid that some use cases might break. Any thoughts …?

**Comment by Johennes** [comment]
Yeah, that's a pickle. :thinking: 

You could maybe argue that `#` is a control character that shouldn't be searched for. For comparison, it doesn't seem to be possible to search for `"` or `"foo` either in the global search.

If the global search supported regexes, you could make `#foo` searches be tag-only and let users resort to `/#foo/` if they really need a substring search.

For me, I think the keywords workaround should suffice for now because at least currently I don't have a use for inline tags.

**Comment by nathanlesage** [comment]
> You could maybe argue that # is a control character that shouldn't be searched for. For comparison, it doesn't seem to be possible to search for " or "foo either in the global search.

That's true and that is my thinking as well. I'll think about this and if I don't come up with any reasonable scenario for keeping `#` as an actual character if it's the beginning of a search term, I'll implement it like that. After all, that should be what most people expect how it behaves.
